src/main/java/Duke.java

```diff
@@ -1,10 +1,217 @@
+        String at = line.replaceAll(GET_AT_REGEX, "");
+
+        // Create new Event instance an add it to end taskList
+        taskList[taskCount] = new Event (description, at);
+        taskCount = taskCount + 1;
+
+        printAcknowledgement();
+    }
+
+    // Print list of task
+    public static void printList() {
+        printDashLine();
+
+        System.out.println("Here are the tasks in your list");
+
+        // Print each item
+        for (int i = 0; i < taskCount; ++i) {
+            System.out.print(i + 1 + ".");
+            taskList[i].printTask();
+        }
+
+        printDashLine();
+    }
+
+    // Mark the task at the given index as done
+    public static void markAsDone(int index){
+        // Check if task is already done
+        if (taskList[index].isDone()){
+            printDashLine();
+            System.out.println("The task has already been completed.");
+            printDashLine();
+            return;
+        }
+
+        // Mark the index as done
+        taskList[index].markAsDone();
+
+        // Acknowledge task is done
+        printDashLine();
+        System.out.print("Nice! I've marked this task as done:\n  ");
+        taskList[index].printTask();
+        printDashLine();
+    }
+
+    // Reads the input of the line to determine the command and run it
+    public static void readInput(String line){
+
+        // Get command from the line
+        String[] splitLine = line.split(" ", 2);
+        String command = splitLine[0];
+
+
+        // Remove command from line
+        line = line.replaceAll(START_LINE_REGEX + command + SPACE_REGEX, "");
+
+        // Check the command type then execute the commands
+        if(line.equals(LIST)) {
+            printList();
+        } else if (command.equals(DONE) && line.matches(DIGITS_REGEX)){
+            // Check if the command is done and is followed by a number
+            int index = Integer.parseInt(line) - 1;
+
+            // Only do markAsDone if the index is within the range of number of tasks
+            if (index < taskCount){
+                markAsDone(index);
+            } else {
+                printError();
+            }
+        } else if (command.equals(TODO)){
+            addTodo(line);
+        } else if (command.equals(DEADLINE) && line.matches(DEADLINE_REGEX)){
+            // Check if line contains "/by"
+            addDeadline(line);
+        } else if (command.equals(EVENT)&& line.matches(EVENT_REGEX)){
+            // Check if line contains "/at"
+            addEvent(line);
+        } else {
+            //  Print Error if the input does not fulfil any of the functions
+            printError();
+        }
+    }
+
+    // Continuously take in input and running the commands until input is BYE then exit
+    public static void runDuke(){
+        // Initialise variables to get input from user
+        String line;
+        Scanner in = new Scanner(System.in);
```

Comment:
Perhaps a more descriptive naming for this would be better? like userInput etc. And I noticed some of your comments are explained in the method name already, would it be more appropriate to remove them in that case? But otherwise it looks it looks good to me! :)